charts/retool/templates/deployment_backend.yaml

```diff
@@ -77,21 +77,33 @@ spec:
             value: {{ template "retool.deploymentTemplateVersion" . }}
           - name: NODE_ENV
             value: production
-          {{- if include "retool.jobRunner.enabled" . }}
-          {{ if ( not $.Values.dbconnector.java.enabled ) }}
-          - name: DISABLE_JAVA_DBCONNECTOR
-            value: "true"
-          {{ end }}
+          {{- /*
+            We are making an assumption here that if we see an invalid tag, we are running a relatively recent version of Retool.
+            Older versions of Retool should always have a semver-compatible tag; only internal versions should be noncompliant.
+          */}}
+          {{- $valid_retool_version_regexp := "([0-9]+]\\.[0-9]+(\\.[0-9]+)?(-[a-zA-Z0-9]+)?)" }}
+          {{- $retool_version_with_java_dbconnector_opt_out := ( or ( not ( regexMatch $valid_retool_version_regexp $.Values.image.tag ) ) ( semverCompare ">= 3.93.0-0" ( regexFind $valid_retool_version_regexp $.Values.image.tag ) ) ) }}
```

Comment:
reading this is tough in line, is there a helper file we can add this to lol

Reply:
an example of a helm function: https://github.com/tryretool/retool-k8s/blob/e053bbf66ad991dbed245adf745aa4605cded79d/terragrunt/src/managed-self-hosted/helm/retool-fork/charts/postgresql/charts/common/templates/_ingress.tpl#L36-L42 we can put it in a file called `_helpers.tpl`